tests/arch/arm/arm_custom_interrupt/testcase.yaml

```diff
@@ -7,4 +7,5 @@ common:
   arch_allow: arm
 tests:
   arch.arm.custom_interrupt:
-    filter: not CONFIG_TRUSTED_EXECUTION_NONSECURE
+    filter: not CONFIG_TRUSTED_EXECUTION_NONSECURE and
+            not dt_compat_enabled("renesas,ra-interrupt-controller-unit")
```

Comment:
Please don't.

If the tests has some requirements that the platform does meet have it reflected in the configuration system. Do not add vendor specific exclusion in non-vendor specific test cases.

See https://github.com/zephyrproject-rtos/zephyr/pull/74444 for a similar case.

Reply:
In this case, renesas-ra-icu is a custom interrupt controller, so I want to exclude the CONFIG_ARM_CUSTOM_INTERRUPT_CONTROLLER=y case, but the test makes it to y with prj.conf.
As a result, if excluding CONFIG_ARM_CUSTOM_INTERRUPT_CONTROLLER=y, skip all cases.
In this situation, I think it would be difficult to do anything other than the current method.

In the case of #74444, even though GEN_ISR_TABLES was set to n in prj.conf, the GEN_ISR_TABLES enabled by select was set to y, so it was successfully excluded.

Reply:
So, if I understand correctly, this implements a custom interrupt controller but excludes it from running the validation tests ?

That sounds like the opposite of what we'd want.

Reply:
This test seems to verify that `The interrupt functions can correctly override if ARM_CUSTOM_INTERRUPT_CONTROLLER switch to on.`
Since I'm assuming that CUSTOM_CONTROLLER is not enabled, I think it is not suited for devices that are already 
The custom interrupt controller itself is tested in other tests. The focus here seems to be on whether `it can be customized`.

Reply:
After further consideration, I decided not to use CONFIG_ARM_CUSTOM_INTERRUPT_CONTROLLER in this PR. It's no longer a problem. Sorry for bothering you.